### PR TITLE
README.md: Improve contribution guide.

### PR DESCRIPTION
**Pull request by gurpreetsingh00885**
The guide for contributing to the pclubuiet website has been improved.
Steps have been added to make sure newcomers can easily contribute.

### CODE REVIEW

**Comment by gurpreetsingh00885** [comment]
@divyam3897 the 'get issue assigned' step would require the contributor to be a member of this organisation. 
This needs a little bit of discussion.

File: README.md [hunk 1]
```diff
@@ -1,26 +1,92 @@
...
+5. Install the dependencies for the project.
+    ```
+    (env) $ cd website
+    (env) $ pip3 install -r requirements.txt
+    ```
+    
+6. Run the migrations
+    ```
+    (env) $ python manage.py migrate
+    ```
```
Comment (divyam3897) [inline README.md]: python manage.py collect statiic? Why did you remove that?

File: README.md [hunk 2]
```diff
@@ -1,26 +1,92 @@
...
+   
+   Once the issue is assigned to you, you can start working on it. Step 10 and beyond will guide you through this part of contribution.
+   
+   **Only PR from the assignee would be considered. This is to save yours and ours time and energy.**
+   **No two contributors shall be working on the same issue.**
+   
+10. Create a new branch and switch to it. (make sure you are on master before doing this).
+    ```
+    (env) $ git branch mybranch
+    (env) $ git checkout mybranch
```
Comment (divyam3897) [inline README.md]: can be done in a single step using "git checkout -b mybranch"

File: README.md (hunk 1 above)
Comment (gurpreetsingh00885) [reply README.md]: Forgot that. My bad.

File: README.md [hunk 3]
```diff
@@ -1,26 +1,92 @@
...
+13. Then create a PR from that branch using GitHub.
 
-Install the dependences
+**What after you have submitted a PR?**
 
-    pip install -r requirements.txt
+Well, you could wait for it to be reviewed by someone or you could attempt to fix another issue. 
 
-Run the migrations and collect the static files
+*OR...*
```
Comment (divyam3897) [inline README.md]: change it to "OR"

File: README.md (hunk 2 above)
Comment (gurpreetsingh00885) [reply README.md]: This guide is for newcomers. So I thought it must be basic.

File: README.md (hunk 3 above)
Comment (gurpreetsingh00885) [reply README.md]: 👍

File: README.md [hunk 4]
```diff
@@ -1,26 +1,92 @@
 # Programming club UIET #
 This is the official website of Programming club, UIET, Panjab University
 
-## Quickstart ##
+## Wanna contribute? here's how ##
```
Comment (divyam3897) [inline README.md]: should be changed to something to "Set up your dev environment" or something better? See https://github.com/matiassingers/awesome-readme

File: README.md [hunk 5]
```diff
@@ -1,26 +1,92 @@
...
+    $ virtualenv env
+    ```
+    We recommend using python-virtualenv. Any other packages would do fine though.
 
-    git clone https://github.com/pclubuiet/website.git
+3. Activate the newly created virtual env:
+    ```
+    $ cd env
+    $ source bin/activate
+    ```
```
Comment (divyam3897) [inline README.md]: I think we should skipp the $ it's not the same for everybody and we should just keep the commands.

File: README.md [hunk 6]
```diff
@@ -1,26 +1,92 @@
...
+    On successful activation, the prompt would change from `$ ` to `(env) $ `.
```
Comment (divyam3897) [inline README.md]: same here, skip this part as it depends on what theme/shell a person is using

Comment (gurpreetsingh00885) [reply README.md]: okay

File: README.md (hunk 2 above)
Comment (divyam3897) [reply README.md]: let's keep it short and simple.

**Comment by gurpreetsingh00885** [comment]
Requested changes have been made How does it look now? @divyam3897

File: README.md [hunk 7]
```diff
@@ -1,26 +1,99 @@
 # Programming club UIET #
 This is the official website of Programming club, UIET, Panjab University
 
-## Quickstart ##
+## Setting up your Development Environment ##
+1. First fork this repo to your account.
```
Comment (divyam3897) [inline README.md]: "," after First?

File: README.md [hunk 8]
```diff
@@ -1,26 +1,99 @@
...
+    ```
+    'mybranch' can be replaced by your preferred name for the branch.
+    The above to commands are equivalent to the following
+    ```
+    git checkout -b mybranch
+    ```
 
-cd to the repository
+11. Make your changes. Then stage them and commit them.
+    See coala's guide to writing good commit messages [here](http://api.coala.io/en/latest/Developers/Writing_Good_Commits.html).
```
Comment (divyam3897) [inline README.md]: Check this https://chris.beams.io/posts/git-commit/ and add this one (I won't prefer adding the guidelines of a particular org here).

**Comment by gurpreetsingh00885** [comment]
Some changes do require multiple commits. So Squashing is not always required. What do you say?

**Comment by divyam3897** [comment]
It depends, if you are working on a single feature it is to be done in a single commit IMHO but if your work consists of multiple features then IMHO you can go for multiple commits (one commit per feature) and as if one works on one issue -> one feature then they should keep it to one commit.

File: README.md [hunk 9]
```diff
@@ -1,26 +1,100 @@
 # Programming club UIET #
 This is the official website of Programming club, UIET, Panjab University
 
-## Quickstart ##
+## Setting up your Development Environment ##
+1. First, fork this repo to your account.
```
Comment (divyam3897) [inline README.md]: umm, use repository everywhere. Let's keep it "Fork this repository..." and similarly for all steps also if not done the same way :)

**Comment by divyam3897** [comment]
It has been merged, thank you
